onnxruntime/core/providers/webnn/builders/model_builder.cc

```diff
@@ -97,8 +97,8 @@ Status ModelBuilder::RegisterInitializers() {
     const auto& shape = tensor.dims();
     std::vector<int32_t> dims;
     if (shape.empty()) {
-      // This is a scalar initializer, WebNN requires a shape, make this a {1} tensor.
-      dims = {1};
+      // This is a scalar initializer.
```

Comment:
Do you actually need the special `if` branch, since the general else branch will do no work anyway (cend == cbegin) in that case?

Reply:
Just want to inform readers that what a scalar is presented in WebNN EP. I could remove `if` branch and add additional comment to inform this.